stdlib/LinearAlgebra/src/eigen.jl

```diff
@@ -275,6 +275,17 @@ function eigen(A::AbstractMatrix{T}; permute::Bool=true, scale::Bool=true, sortb
     isdiag(AA) && return eigen(Diagonal(AA); permute=permute, scale=scale, sortby=sortby)
     return eigen!(AA; permute=permute, scale=scale, sortby=sortby, jvl=jvl, jvr=jvr, jce=jce, jcv=jcv)
 end
+function eigen(A::AbstractMatrix{T}; permute::Bool=true, scale::Bool=true, sortby::Union{Function,Nothing}=eigsortby) where {T <: Union{Float16,Complex{Float16}}}
+    AA = copy_oftype(A, eigtype(T))
+    isdiag(AA) && return eigen(Diagonal(AA); permute=permute, scale=scale, sortby=sortby)
+    A = eigen!(AA; permute=permute, scale=scale, sortby=sortby)
+    values = convert(AbstractVector{isreal(A.values) ? Float16 : Complex{Float16}}, A.values)
+    vectors = convert(AbstractMatrix{isreal(A.vectors) ? Float16 : Complex{Float16}}, A.vectors)
+    vectorsl = convert(AbstractMatrix{isreal(A.vectors) ? Float16 : Complex{Float16}}, A.vectorsl)
+    rconde = convert(Vector{isreal(A.rconde) ? Float16 : Complex{Float16}}, A.rconde)
+    rcondv = convert(Vector{isreal(A.rcondv) ? Float16 : Complex{Float16}}, A.rcondv)
```

Comment:
If these are condition numbers as their names suggest, then these are always real (as also suggested by the default values in the general constructor) and you can convert `A.rconde` and `A.rcondv` to `Float16` directly, without the branch.